New directory is created even if package name is invalid

If you try and create a new repo and its name is invalid, an empty directory is still created at the given path.

e.g.:
```
$ python3-pip-skeleton new /tmp/invalid-package-name --org TEST
...
...
AssertionError: 'invalid-package-name' is not a valid python package name
...
$ ls /tmp/
drwxrwxr-x.   2 <uid> <uid>        4096 Jan 25 10:53 invalid-package-name
```

--- a/src/python3_pip_skeleton/__main__.py
+++ b/src/python3_pip_skeleton/__main__.py
@@ -147,6 +147,8 @@ def verify_not_adopted(root: Path):
 def new(args):
     path: Path = args.path
 
+    package = validate_package(args)
+
     if path.exists():
         assert path.is_dir() and not list(
             path.iterdir()
@@ -154,7 +156,6 @@ def new(args):
     else:
         path.mkdir(parents=True)
 
-    package = validate_package(args)
     git("init", "-b", "main", cwd=path)
     print(f"Created git repo in {path}")
     merge_skeleton(

